internal/literals/numbers.go

```diff
@@ -0,0 +1,277 @@
+package literals
+
+import (
+	"encoding/binary"
+	"errors"
+	"go/ast"
+	"go/token"
+	"go/types"
+	"math"
+	"strconv"
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+func obfuscateNumberLiteral(cursor *astutil.Cursor, info *types.Info) error {
+	// TODO: Add support for obfuscation of hexadecimal and octal notation
```

Comment:
strconv.ParseInt with base 0 may be suitable https://golang.org/pkg/strconv/#ParseInt.

From the documentation, it should automatically recognize the type of number